driver/src/main/java/org/neo4j/driver/internal/util/DriverInfoUtil.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.driver.internal.util;
+
+import static java.lang.String.format;
+
+import java.util.Optional;
+import org.neo4j.driver.Session;
+import org.neo4j.driver.internal.BoltAgent;
+
+public class DriverInfoUtil {
+    public static BoltAgent boltAgent() {
+        var productInfo = format("neo4j-java/%s", driverVersion());
+
+        var platformBuilder = new StringBuilder();
+        getProperty("os.name").ifPresent(value -> append(value, platformBuilder));
+        getProperty("os.version").ifPresent(value -> append(value, platformBuilder));
+        getProperty("os.arch").ifPresent(value -> append(value, platformBuilder));
+
+        var language = getProperty("java.version").map(version -> "Java/" + version);
+
+        var languageDetails = language.map(ignored -> {
+            var languageDetailsBuilder = new StringBuilder();
+            getProperty("java.vm.vendor").ifPresent(value -> append(value, languageDetailsBuilder));
+            getProperty("java.vm.name").ifPresent(value -> append(value, languageDetailsBuilder));
+            getProperty("java.vm.version").ifPresent(value -> append(value, languageDetailsBuilder));
+            return languageDetailsBuilder.isEmpty() ? null : languageDetailsBuilder;
+        });
+
+        return new BoltAgent(
+                productInfo,
+                platformBuilder.isEmpty() ? null : platformBuilder.toString(),
+                language.orElse(null),
+                languageDetails.isEmpty() ? null : languageDetails.toString());
+    }
```

Comment:
Might worth to add some unit tests for checking if the formats are the ones expected.